### ISSUE TITLE
v3.1.0 requires significant more methods in release build

### ISSUE THREAD

**Issue by orionlee** [leinardi/FloatingActionButtonSpeedDial#136]
(It is not necessarily a problem per-se, but a yellow flag for `v3.1.0`.)

### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of the library.

### Step 2: Describe your environment
 - Library version: `3.1.0`
 - Support library version: androidX appcompat `1.1.0`

### Step 3: Describe the problem:

The app, AntennaPod, has recently tried to upgrade FAB Speed Dial library to `v3.1.0` . (https://github.com/AntennaPod/AntennaPod/pull/3468#issuecomment-536388983)

With `v3.1.0`, we noticed the release build required significantly more methods (about 2,000 more, primarily from `com.google.android.material` packages), comparing to `v3.0.0` .

In our case, it made the app exceeded 64K limit of methods in a single dex. It'd be great if our app does not need to become multi-dex because of using the library.

Method Count breakdown of the app.

|   | FAB 3.0.0 | FAB 3.1.0 |
| --- | --- | --- |
|  Total | 63,995 | 66,300 |
|  android + google | 38,034 | 40,315 |
|  support / androidX | 18,257 | 18,573 |
|  com.google | 10,626 | 12,456 |
|  **com.google.android.material** |  2,731 | **4,561** |
|  android platform | 9,151 | 9,286 |
| FAB SpeedDial | 247 | 259

**Comment by leinardi** [comment]
Hi @orionlee, I am very sorry that the latest version of SpeedDial is causing you issues, I'm a very happy user of AntennaPod.
Sadly, as you highlighted already on the table above, the new methods are not in the SpeedDial code, but they come from the version 1.1.x of com.google.android.material.
One way to [work around](https://stackoverflow.com/questions/28444016/how-can-i-force-gradle-to-set-the-same-version-for-two-dependencies) this issue for you is to force com.google.android.material to version 1.0.0:
```
configurations.all {
    resolutionStrategy {
        force 'com.google.android.material:material:1.0.0'
    }
}
```
But this is just delaying the inevitable: soon the new version will become the one you have to use to if you want to have the new official material component widgets.

I have one question for you: could you tell me where in AntennaPod is the SpeedDial used? I can't find it in any of the main screens.

**Comment by orionlee** [comment]
Thanks for looking into it. 

In AntennaPod, the SpeedDial has recently been used in batch edit screens (of a podcast's episode listing, queue, etc.).

**Comment by leinardi** [comment]
Hi @orionlee, I just released a new version, 3.1.1, that is reverting the switch to material component 1.1.0. I'm closing this issue since the problem doesn't exist anymore (until I'll switch back to 1.1.0 once it get released as stable).
